docs/src/modules/results.md

```diff
@@ -0,0 +1,37 @@
+# Results
+
+Results are an idea taken from
+[`rust`](https://doc.rust-lang.org/rust-by-example/error/result.html). The main idea is to have
+types that can be used to return values and provide ways to handle errors if they ocurr.
+
+```solidity
+import { Test, StringResult, Ok, console } from "vulcan/test.sol";
+
+contract TestMyContract is Test {
+    function testMyContract() external {
+        // foo.bar() returns a StringResult
+        StringResult result = foo.bar();
+
+        // Check if the function failed
+        if (result.isError()) {
+            // Decode the error and get the message that can explain why the function failed
+            (, string memory message, ) = result.toError().decode();
```

Comment:
the correct way of using errors is to use the `decodeAs` functions, such as:
```solidity
Error err = result.toError();
if (err.matches(SomeErrorFunc)) {
    uint256 val = err.decodeAs(SomeErrorFunc)
}
```

As seen here, "decoding" an error should only extract the encoded error data. The message is not accessible from the outside. We could do it by adding a message function or something if we need it, however I don't see it being necessary because there are 2 usecases:
1. Throwing the error (result.unwrap() or result.expect(...))
2. Handling the error (using err.matches(SomeErrorFunc))

Only in `result.unwrap()` the default message is relevant.

Reply:
An improvement to the example could be something like (imo all examples should be testable by directly cp/pasting them):

```solidity
CommandResult result = commands.run(["echo", "Hello, world!"]);

// We could immediately get the actual output, reverting if the command failed to run
bytes memory out = result.expect("wtf echo failed!").stdout;

// Or we can handle different types of errors
if (result.isError()) {
    Error err = result.toError();

    if (err.matches(CommandError.NotExecuted)) {
        revert("Something weird happened!");
    } 

    revert("Unknown error");
}
```